---
title: "Figure 1"
date: "2024-07-03"
output: 
  github_document:
    toc: true
    toc_depth: 2
---

```{r setup, include=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load libraries

```{r, warning=FALSE, eval=TRUE, message=FALSE}
.libPaths( c( "/data/Common_Folder/R/Single_cell_packages/", .libPaths()) )

library(Seurat)
library(ggplot2)
library(dplyr)
library(data.table)
library(speckle)
library(openxlsx)
library(data.table)
library(dplyr)
library(openxlsx)

# Uncomment the line below to install the 'presto' package if needed 
#devtools::install_github('immunogenomics/presto')
```

## Load the RNA-Seq data

```{r}
# Load the processed Seurat object from the specified path
pd = readRDS("/data/dehestani/scRNAseq_iOligo/Processed_object/ODC35_Seurat_SCT_harmony_annotated_v1.rds")
pd
```

## Adding cluster cell type names

```{r}
# Assign cell type names to clusters
Idents(pd) <- "sub.cluster"
new.cluster.ids <- c("iPPC","iNL2", "iOPC", "iCEP", "iNL1", "iRGC", "iODC", "iINPC")
names(new.cluster.ids) <- levels(pd)
pd <- RenameIdents(pd, new.cluster.ids)
pd@meta.data$BroadCellType <- as.character(Idents(pd))

# Display the count of cells in each cell type
table(pd$BroadCellType)
```

## Cell Composition Analysis

```{r}

# Extract metadata and convert to data.table
md <- pd@meta.data %>% as.data.table()

# Calculate cell numbers and percentages for each cluster
cell_counts_f <- md[, .N, by = c("BroadCellType")]
total_cells <- sum(cell_counts_f$N)
cell_counts_f$Percentage <- (cell_counts_f$N / total_cells) * 100

# Calculate cell numbers and percentages for each cluster and mutation
cell_counts_fm <- md[, .N, by = c("BroadCellType", "Mutation")]
total_cells <- sum(cell_counts_fm$N)
cell_counts_fm$Percentage <- (cell_counts_fm$N / total_cells) * 100

# Pivot table to get LRRK2 and HC percentages
cell_counts_pivot <- dcast(cell_counts_fm, BroadCellType ~ Mutation, value.var = "N")
cell_counts_pivot[, `:=`(LRRK2_Percentage = (LRRK2 / total_cells) * 100, HC_Percentage = (HC / total_cells) * 100)]

# Combine the data into the final table
final_table <- merge(cell_counts_f, cell_counts_pivot, by = "BroadCellType")
final_table <- final_table[, .(BroadCellType, N, Percentage, LRRK2, LRRK2_Percentage, HC, HC_Percentage)]

# Rename columns to match the desired structure
colnames(final_table) <- c("BroadCellType", "N", "Percentage", "LRRK2", "Percentage", "HC", "Percentage")

final_table


```

```{r, include=FALSE}
# Create a new workbook and add a worksheet
#wb <- createWorkbook()
#addWorksheet(wb, "Sheet1")

# Write the data to the worksheet
#writeData(wb, "Sheet1", final_table, startRow = 3, startCol = 2)

# Add bold titles with the same font size and style
#header_style <- createStyle(fontSize = 12, fontName = "Calibri", textDecoration = "bold")
#addStyle(wb, "Sheet1", header_style, rows = 3, cols = 2:8, gridExpand = TRUE)

# Define the path to save the output table and save the workbook
#pathto.outTable <- "/data/nasser/Manuscript/table/Supplementary Tables/"
#saveWorkbook(wb,  file = paste0(pathto.outTable,"Sup2_incls8.xlsx"), overwrite = TRUE)

```

## Figure 1-C: UMAP visualisation

```{r}
# Set cell type identities
Idents(pd) <- "BroadCellType"

# Define output path for plots
#pathto.outPlots <- "/data/nasser/Manuscript/plots/figure1/"
#outName <- "Figure1_b_"

# Define color palette for UMAP plot
rushmore_palette <- c("#6677B3", "#99CC66", "#66C2A5", "#FFCC66", "#FF9933", "#FF9999", "#C85200", "#B3B3B3", "#999999", "#666666")

# Create and save UMAP plot
#pathto.outPlots = "/data/nasser/Manuscript/plots/figure1/"
#png(paste0(pathto.outPlots,outName,"UMAP_incl_sample8.png"), width=1500, height=1000,res = 300)
p=DimPlot(pd, reduction = "umap", label = T, pt.size = 0.5, cols = rushmore_palette) 
#dev.off()

# Display the UMAP plot
p

```

## Excluding sample 8 from the analysis

```{r}
# Define output path for processed object
#pathto.outData = "/data/nasser/Manuscript/processedobject/"

# Exclude sample 8 and create a new subset object
Idents(pd) <- "SampleID"
pd <- subset(pd,idents = c("s1", "s2", "s3", "s4", "s5", "s6", "s7") )

# Save the new subset object
#saveRDS(pd, file = paste0(pathto.outData, "ODC35_woClus8_nasser"))
```

##Read in object without cluster 8

```{r, include=FALSE}
# Load the processed Seurat object without sample 8
pd = readRDS("/data/nasser/Manuscript/Strict_threshold/processedobject/ODC35_woClus8_subclust3_res0.15_NK")
# Assign cell type names to clusters
Idents(pd) <- "BroadCellType"
new.cluster.ids <- c("iPPC","iNL2", "iOPC", "iCEP", "iNL1", "iRGC", "iODC", "iINPC")
names(new.cluster.ids) <- levels(pd)
pd <- RenameIdents(pd, new.cluster.ids)
pd@meta.data$BroadCellType <- as.character(Idents(pd))


```

## Figure 1-G :Dotplot with known marker genes for cluster identification

```{r, include=TRUE, warning=FALSE}
# Define output path for plots
#pathto.outPlots <- "/data/nasser/Manuscript/plots/figure1/"
#outName <- "Figure1_d_"

# Set the default assay to RNA
DefaultAssay(pd) <- "RNA"

# Set identities to BroadCellType
Idents(pd) = "BroadCellType"

# Define marker genes for dot plot
Genes_dotPlot <- rev(unique(c( "GFAP", "NES","BTG2","HES6", "RBFOX3", "SLC17A6", "SNAP25","RIT2" ,"TH", "ZCCHC12", "SNCA", "CLDN5", "SHH","TOP2A", "MKI67","LRRK2","PDGFRA","MAG", "MBP", "MOG", "PLP1","CNP" )))

# Set levels for pd.sub
levels(pd) <- c('iRGC','iINPC','iNL1', 'iNL2', 'iCEP','iPPC','iOPC','iODC')

# Dot plot for cell markers
#png(paste0(pathto.outPlots,outName,"Dotplot_cellmarkers_update.png"), width=2500, height=1500, res = 300)
DotPlot(pd, idents = c('iRGC','iINPC','iNL1', 'iNL2', 'iCEP','iPPC ','iOPC','iODC'), assay = "RNA", features = Genes_dotPlot,  dot.min = 0.1, cols=c("blue", "red")) + scale_colour_gradient2(low="steelblue", mid="lightgrey", high="red") + RotatedAxis()
#dev.off()


```

```{r, include=FALSE}
#png(paste0(pathto.outPlots,outName,"Featureplot_cellmarkers_lrrk2_pdgfra_scna_th.png"), width=2500, height=1500, res = 300)
#FeaturePlot(pd, features = c("LRRK2", "PDGFRA", "SNCA", "TH"))
#dev.off()
```

```{r, include=FALSE}
## iOPC exploration of LRRK2, RIT2, PDGFRA genes

# Define output name for plots
#outName <- "Figure1_f_"

# Subset for iOPC cell type
celltype_subset <- subset(pd, idents = "iOPC")
Idents(celltype_subset) <- "Mutation"

# Create and save dot plot for iOPC
#png(paste0(pathto.outPlots,outName,"Dotplot_iOPC_condition.png"), width=2000, height=1100, res = 300)
DotPlot(celltype_subset, features = c("LRRK2", "RIT2", "PDGFRA"), cols=c("blue", "red")) + scale_colour_gradient2(low="steelblue", mid="lightgrey", high="red") + RotatedAxis()
#dev.off()

# Create and save violin plot for iOPC
#png(paste0(pathto.outPlots,outName,"vlnplot_iOPC_condition.png"), width=2000, height=1100, res = 300)
VlnPlot(celltype_subset, features = c("LRRK2", "RIT2", "PDGFRA"))
#dev.off()

```

```{r, include = FALSE}
## Cluster iNL2 expression per condition

# Define output name for plots
#outName <- "Figure1_f_"

# Subset for iNL2 cell type
celltype_subset <- subset(pd, idents = "iNL2")
Idents(celltype_subset) <- "Mutation"

# Create and save dot plot for iNL2
#png(paste0(pathto.outPlots,outName,"Dotplot_iNL2_condition.png"), width=2000, height=1100, res = 300)
DotPlot(celltype_subset, features = c("SNCA"), cols=c("blue", "red")) + scale_colour_gradient2(low="steelblue", mid="lightgrey", high="red") + RotatedAxis()
#dev.off()

# Create and save violin plot for iNL2
#png(paste0(pathto.outPlots,outName,"vlnplot_iNL2_condition.png"), width=2000, height=1100, res = 300)
VlnPlot(celltype_subset, features = c("SNCA"))
#dev.off()

```

## Figure 1-F: Celltype composition calculation

```{r}

# Extract metadata and convert to data.table

md <- pd@meta.data %>% as.data.table()

# Calculate cell numbers and percentages for each cluster

cell_counts <- md[, .N, by = c("BroadCellType")]

total_cells <- sum(cell_counts$N)

cell_counts$Percentage <- (cell_counts$N / total_cells) * 100

# Calculate cell counts across samples split by mutation

h <- subset(pd, subset = LRRK2 > 0)

md <- pd@meta.data %>% as.data.table()

md[, .N, by = c("BroadCellType")]

d <- md[, .N, by = c("SampleID", "BroadCellType")] %>% dcast(., SampleID ~ BroadCellType, value.var = "N")

# Summarize cell counts and calculate percentages for each cell type

d_sum <- d %>%

  rowwise() %>%

  mutate(sum = sum(iCEP, iOPC, iODC, iPPC , iRGC, iINPC, iNL1, iNL2, na.rm = T))

d_sum$iCEP <- (d_sum$iCEP / d_sum$sum * 100)

d_sum$iOPC <- (d_sum$iOPC / d_sum$sum * 100)

d_sum$iODC <- (d_sum$iODC / d_sum$sum * 100)

d_sum$iPPC <- (d_sum$iPPC / d_sum$sum * 100)

d_sum$iRGC <- (d_sum$iRGC / d_sum$sum * 100)

d_sum$iINPC <- (d_sum$iINPC / d_sum$sum * 100)

d_sum$iNL1 <- (d_sum$iNL1 / d_sum$sum * 100)

d_sum$iNL2 <- (d_sum$iNL2 / d_sum$sum * 100)

# Add mutation label according to the subjects order in the metadata

d_sum$Mutation <- c("LRRK2", "LRRK2", "LRRK2", "HC", "HC", "HC", "HC")

d_sum

```

```{r, include=FALSE}
## Celltype composition boxplot

# Define output name for plots
#outName <- "Figure1_g_"

# Create and save a boxplot for cell type composition
#png(paste0(pathto.outPlots,outName,"celltype_composition_boxplot.png"), width=2000, height=1100, res = 300)
ggplot(pd@meta.data, aes(x=CellType, fill=Mutation)) + 
  geom_bar(position = "fill") + geom_text(stat = 'count', aes(label = ..count..), position = position_fill(vjust = 0.5)) + RotatedAxis() + theme_classic()
#dev.off()


```

```{r, include= FALSE}
#Speckle analysis results

# Calculate Fisher's exact test on cell type mutation composition to find the difference in the distribution of cell counts across different cell types (clusters) between the "HC" (Healthy Control) and "LRRK2" group

# Define the path to save the output table
#pathto.outTable <- "/data/nasser/Manuscript/table/Supplementary Tables/"

# Perform Speckle analysis for confounding testing
Idents(pd) <- "CellType"
p = propeller(clusters=pd$CellType, sample=pd$SampleID, group=pd$Mutation)

p

# Save the results to a CSV file
#write.table(p, file = paste0(pathto.outTable,"S2_speckle_old.csv"), quote = F, sep = ",", row.names = T)

```
